kolena/_utils/batched_load.py

```diff
@@ -79,6 +82,26 @@ def upload_data_frame_chunk(df_chunk: pd.DataFrame, load_uuid: str) -> None:
 DFType = TypeVar("DFType", bound=LoadableDataFrame)
 
 
+def upload_data_frame_in_smart_chunks(df: pd.DataFrame, uuid: str, rows: int = 1000) -> None:
+    df_memory = df.memory_usage(index=True, deep=True).sum()
+    batch_size = (
+        len(df)
+        if df_memory <= SHORT_CIRCUT_LIMIT and len(df) > 0
+        else (BATCH_LIMIT // _get_preflight_export_size(df, rows)) * rows
+    )
+    upload_data_frame(df, batch_size, uuid)
+
+
+def _get_preflight_export_size(df: pd.DataFrame, rows: int) -> int:
+    df_subset = df[:rows]
+    with tempfile.NamedTemporaryFile() as temp:
```

Comment:
I think we can use `io.ByteIO` here, not incurring disk penalty

Reply:
done